stdlib/parser/dsl.md

```diff
@@ -0,0 +1,197 @@
+# Introduction
+
+This file gives a basic runthrough of the in-progress design of a DSL
+for syntax and AST specification.
+
+## Basics
+
+```
+-- Line comments
+
+/-
+Multiline comments
+-/
+```
+
+We declare new types in the AST and non-terminals in
+the grammar with `type`.
+
+```
+type Expr
+```
+
+A constructor in the AST and production in the grammar is introduced
+with `prod`.
+
+```
+prod TmInt: Expr = value:Int
+```
+
+Each constructor gets its own language fragment in the generated code,
+thus the above yields the following fragment:
```

Comment:
Maybe we optionally should make it possible to also just generate one language fragment. It will definitely be more readable.